splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -1135,9 +1146,11 @@ public boolean hasNext() {
                                             } catch (StandardException e) {
                                                 // DB-9890 Skip a column if its statistics object doesn't fit into HBase cell.
                                                 if (e.getCause().getMessage().contains("KeyValue size too large")) {
+                                                    int columnId = nextRow.getColumn(SYSCOLUMNSTATISTICSRowFactory.COLUMNID).getInt();
                                                     SpliceLogUtils.warn(LOG, "Statistics object of [ConglomID=%d, ColumnID=%d] exceeds max KeyValue size. Try increase hbase.client.keyvalue.maxsize.",
                                                             nextRow.getColumn(SYSCOLUMNSTATISTICSRowFactory.CONGLOMID).getInt(),
-                                                            nextRow.getColumn(SYSCOLUMNSTATISTICSRowFactory.COLUMNID).getInt());
+                                                            columnId);
+                                                    skippedColIds.add(columnId);
```

Comment:
For non-merged stats (that is, we output one row in the table stats table for each partition, so are the column stats), that means, we may add the same columnId multiple times to the `skippedColIds` list. Non-merged stats is a mode that we are not using currently, still I prefer to maintain it (in consideration in the future to support incremental stats) and add duplicate checks for this path. You may collect non-merged stats by calling the system procedure `COLLECT_NONMERGED_TABLE_STATISTICS`.

Reply:
I see. I pushed a new commit and use HashSet instead of ArrayList.